src/common/dom/apply_themes_on_element.ts

```diff
@@ -37,6 +38,13 @@ export const applyThemesOnElement = (
     if (themeOptions.dark) {
       cacheKey = `${cacheKey}__dark`;
       themeRules = darkStyles;
+      if (themeOptions.primaryColor) {
+        themeRules["app-header-background-color"] = blend(
+          themeOptions.primaryColor,
+          "#111",
```

Comment:
Let's not expand if we don't have to
```suggestion
          "#111111",
```